### PR TITLE
format(expr) beautification for quantifiers and = on booleans

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: fa434a597a62f4e3658b3ee058d39ace43ab0f3b).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85166064
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 3629a4ae57914d12fb16245efd8445577ac7b383).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85169864

File: src/util/format_expr.cpp [hunk 1]
```diff
@@ -80,6 +80,13 @@ static std::ostream &format_rec(std::ostream &os, const multi_ary_exprt &src)
     operator_str = u8"\u21d2"; // =>, U+21D2
   else if(src.id() == ID_iff)
     operator_str = u8"\u21d4"; // <=>, U+21D4
+  else if(src.id() == ID_equal)
+  {
+    if(!src.operands().empty() && src.op0().type().id() == ID_bool)
+      operator_str = u8"\u21d4"; // <=>, U+21D4
```
Comment (smowton) [inline src/util/format_expr.cpp]: I think this would confuse someone using the output to debug, since they'd go looking for ID_iff

Comment (kroening) [reply src/util/format_expr.cpp]: Remove ID_iff?

Comment (tautschnig) [reply src/util/format_expr.cpp]: It's used only four times across the code base, that makes for an almost trivial PR.

Comment (kroening) [reply src/util/format_expr.cpp]: #2999

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: be0aa3201dd19aa645b254f6958c46b1c9125cd0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85359340

**Review by tautschnig — CHANGES_REQUESTED** [review]
The fact that user-visible output changes but no changes to tests were necessary implies that we don't actually test this code. This might change if #2977 were merged, so let's get that merged first.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 37f5b8eeb8a29ff7baff60f912cea01a2476f8ff).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85388192

**Review by smowton — APPROVED** [review]

**Comment by kroening** [comment]
@tautschnig This is now somewhat circular: to get #2977 merged, a lot of tests need to pass, and it's easier to achieve that if the output of format() is moved to be more similar to that what's there right now.
This PR was motivated by getting #2977 to merge.

**Review by tautschnig — APPROVED** [review]
Ok, let's add the tests in #2977 then.

**Pull request merged**